src/Mod/PartDesign/Gui/TaskChamferParameters.cpp

```diff
@@ -194,6 +198,31 @@ void TaskChamferParameters::clearButtons(const selectionModes notThis)
     DressUpView->highlightReferences(false);
 }
 
+// By adding the edges to the Gui.Selection in python, then onSelectionChanged() gets called
+// where the edges get added to the list in the task dialog and highlighted in the 3d view
+
+void TaskChamferParameters::onSelectAllEdges(void)
+{
+    PartDesign::Chamfer* pcChamfer = static_cast<PartDesign::Chamfer*>(DressUpView->getObject());
+    if (pcChamfer) {
+        App::DocumentObject* base = pcChamfer->Base.getValue();
+        if (base) {
+            std::string baseName = base->getNameInDocument();
+            std::ostringstream command;
+            command << "from PySide import QtGui,QtCore\n";
+            command << "QtGui.QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)\n";
+            command << "for _ii in range(0, len(App.ActiveDocument.getObject(\'" << baseName << "\').Shape.Edges)):\n";

```

Comment:
For the recorded Python code do not use App.ActiveDocument because the user can have opened several documents where a document is activated that doesn't belong to the opened dialog. Thus, using App.ActiveDocument will be executed for the wrong document. Instead use DocumentObjectT::getObjectPython() or one of the helper functions defined inside CommandT.h.